test/cpp/api/module.cpp

```diff
@@ -297,6 +293,22 @@ TEST_F(ModuleTest, CloneCreatesDistinctParameters) {
   }
 }
 
+TEST_F(ModuleTest, CloneCreatesDistinctParameters) {
+  auto module = std::make_shared<TestDistinctParametersModule>();
+  torch::NoGradGuard no_grad;
+  auto module2 = module->clone();
+  testDistinctParameters(module, module2);
+}
+
+TEST_F(ModuleTest, CloneCreatesDistinctParametersExplicitDevice_CUDA) {
+  auto module = std::make_shared<TestDistinctParametersModule>();
+  torch::NoGradGuard no_grad;
+  torch::Device device(torch::kCUDA, 0);
+  module->to(device);
+  auto module2 = module->clone(device);
```

Comment:
I think we should also have another test for the case where the `module`'s original device and the clone-to device are different.

Reply:
Hi @yf225, thanks for reviewing and thanks for the catch! Let me add a test.